dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpServlet.java

```diff
@@ -119,6 +120,10 @@ private void callTool(McpTransport transport) {
             Object result = params.getMethod().invoke(bean, params.getArguments(jsonb));
             if (result instanceof ToolResponse response) {
                 transport.sendResponse(response);
+            } else if (result instanceof List<?> list && list.stream().allMatch(item -> item instanceof Content)) {
```

Comment:
I think this line should still have the `!list.isEmpty()` check in it. `allMatch` will return `true` if the list is empty.

If a method returns an empty list, I think we should assume it could have been a list of POJOs (and treat it as structured content) rather than assuming it could have been a list of `Content` objects.